### PR TITLE
add ErrorWriter to Server

### PR DESCRIPTION
**Pull request by rogpeppe**
This makes it possible to return arbitrary error responses, not
just JSON-encoded errors.

### CODE REVIEW

**Review by mhilton — CHANGES_REQUESTED** [review]
I'm yet to be convinced there as a need for this and I'm loath to add more infrastructure here without good reason.

File: handler.go [hunk 1]
```diff
@@ -29,6 +29,13 @@ type Server struct {
 	//
 	// If this is nil, DefaultErrorMapper will be used.
```
Comment (mhilton) [inline handler.go]: If both this and ErrorWriter are nil then, ...

File: handler_test.go [hunk 2]
```diff
@@ -1098,6 +1099,33 @@ var writeErrorTests = []struct {
...
+		},
+		ErrorMapper: func(_ context.Context, err error) (int, interface{}) {
+			return http.StatusInternalServerError, nil
+		},
+	},
+	assertResponse: func(c *gc.C, rec *httptest.ResponseRecorder) {
+		c.Assert(rec.Body.String(), gc.Equals, "custom error")
+		c.Assert(rec.Code, gc.Equals, http.StatusOK)
+	},
 }}
```
Comment (mhilton) [inline handler_test.go:1130]: Adding an example would be really nice. although possibly a bit out of scope.

**Review by mhilton — APPROVED** [review]
LGTM with a couple of thoughts.

File: handler.go (hunk 1 above)
Comment (rogpeppe) [reply handler.go]: done

File: handler_test.go (hunk 2 above)
Comment (rogpeppe) [reply handler_test.go:1130]: yeah, we could do with lots more examples. another PR, I think.

**Pull request merged**